apps/03TicTacToe.js

```diff
@@ -19,36 +19,70 @@ function printBoard() {
     console.log('1 ' + board[1].join(' | '));
     console.log('  ---------');
     console.log('2 ' + board[2].join(' | '));
-}
+};
 
 function horizontalWin() {
-    // Your code here
-}
+  return (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) ||
+  (board [1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) ||
+  (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2]=== playerTurn);
+
+};
 
 function verticalWin() {
-    // Your code here
-}
+  return (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) ||
+  (board [0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) ||
+  (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2]=== playerTurn);
+};
 
 function diagonalWin() {
-    // Your code here
-}
+  return (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) ||
+  (board [0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn);
+
+};
 
 function checkForWin() {
-    // Your code here
+  if (verticalWin() || horizontalWin() || diagonalWin()){
+  printBoard();
+  console.log ("Player" + " " + playerTurn + " " + "Wins!" );
+  cleanBoard();
 }
+  else {
+    return false;
```

Comment:
SOOOOOO CLOSE! One test failing. So if you return false here... we should probably also return true in some cases, yeah?